clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/MoveExecutor.scala

```diff
@@ -0,0 +1,212 @@
+package org.broadinstitute.clio.client.dispatch
+
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.headers.HttpCredentials
+import org.broadinstitute.clio.client.ClioClientConfig
+import org.broadinstitute.clio.client.commands.{ClioCommand, MoveCommand}
+import org.broadinstitute.clio.client.util.IoUtil
+import org.broadinstitute.clio.client.webclient.ClioWebClient
+import org.broadinstitute.clio.transfer.model.TransferIndex
+import org.broadinstitute.clio.util.ClassUtil
+import org.broadinstitute.clio.util.generic.{
+  CaseClassMapper,
+  CaseClassTypeConverter
+}
+import org.broadinstitute.clio.util.model.Location
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class MoveExecutor[TI <: TransferIndex](moveCommand: MoveCommand[TI])
+    extends Executor {
+
+  import moveCommand.index.implicits._
+
+  private val prettyKey = ClassUtil.formatFields(moveCommand.key)
+  val destination: String = moveCommand.destination
+  val name: String = moveCommand.index.name
+
+  override def execute(webClient: ClioWebClient, ioUtil: IoUtil)(
+    implicit ec: ExecutionContext,
+    credentials: HttpCredentials
+  ): Future[HttpResponse] = {
+    for {
+      _ <- Future(verifyCloudPaths(ioUtil))
+        .logErrorMsg("clio-client can only handle cloud operations right now.")
+      existingMetadata <- queryForKey(webClient)
+        .logErrorMsg(s"Could not query the $name. No files have been moved.")
+      _ <- Future(checkDestinationVsExistingPaths(existingMetadata))
+      upsertResponse <- moveFiles(webClient, ioUtil, existingMetadata)
+    } yield {
+      upsertResponse
+    }
+  }
+
+  private def verifyCloudPaths(ioUtil: IoUtil): Unit = {
+    if (moveCommand.key.location != Location.GCP) {
+      throw new Exception(
+        s"Only GCP unmapped ${name}s are supported at this time."
```

Comment:
Not sure what `unmapped` means here.

Also this error seem redundant with the error message on line 34 above. Will both errors be printed to the user?

Reply:
`unmapped` is a leftover from the pre-generic days. It is duplicated, I'll pick one to remove.